### PR TITLE
Run patches on setup

### PR DESCRIPTION
**Pull request by Agent00Ming**

### CODE REVIEW

**Comment by Agent00Ming** [comment]
This builds with #5 but I'm not sure if this is the best way to do this as discussed there.

File: meson.build [hunk 1]
```diff
@@ -80,6 +80,8 @@ else
 	)
 endif
 
+patching = run_command('./patches/apply.sh', capture: true, check: true)
```
Comment (fufexan) [inline meson.build:83]: Does this work as is? I thought `sh -c` was needed.

Comment (Agent00Ming) [reply meson.build:83]: It does work. The [meson docs](https://mesonbuild.com/External-commands.html) mention calling a bash script directly with run_command() so I tried it out

**Review by fufexan — APPROVED** [review]
LGTM.

File: patches/apply.sh [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+	git apply --check --reverse -q -p1 $PATCH
+}
+
+if [[ -n "$PATCHES" ]];
+then
+	# check patch validity and apply, else check if already applied and report on failure
+	echo 'Patches found. Applying...';
+	for PATCH in $PATCHES;
+	do
+		check && apply || check_applied || echo =======\> \'$PATCH\' was not applied;
```
Comment (vaxerski) [inline patches/apply.sh]: the script should die with an error code and not proceed if it can't apply a patch. Preferably error out meson.

Comment (Agent00Ming) [reply patches/apply.sh]: Done:
```
meson.build:83:11: ERROR: Command `/home/ming/documents/git/wlroots-hyprland/./patches/apply.sh` failed with status 1.
```

**Review by vaxerski — APPROVED** [review]
lgtm, thanks!

**Pull request merged**